pos_product_template_combo/static/src/js/ProductScreen.js

```diff
@@ -63,16 +71,73 @@ odoo.define("pos_product_template_combo.ProductScreen", function (require) {
             }
 
             _add_combo_products_to_order(result) {
-                result.payload.forEach(async (product_to_add) => {
-                    const options = {
-                        price: product_to_add.price,
-                        quantity: product_to_add.quantity,
-                        merge: false,
-                    };
-                    if (product_to_add.product) {
-                        const order = this.env.pos.get_order();
-                        order.add_product(product_to_add.product, options);
+                const linesAddedToOrder = [];
+                for (let i = 0; i < result.payload.length; i++) {
+                    const productToAdd = result.payload[i];
+                    const options = this._mountComboProductOptions(productToAdd);
+                    if (productToAdd.product) {
+                        this.order.add_product(productToAdd.product, options);
+                        linesAddedToOrder.push(this.selectedOrderline);
+                    }
+                    if (
+                        this._isDuplicateItemCategoryBehavior(
+                            productToAdd.categoryBehavior
+                        )
+                    ) {
+                        try {
+                            this._insertDuplicateOrderline(
+                                productToAdd,
+                                linesAddedToOrder
+                            );
+                        } catch (error) {
+                            this._rollbackComboOrderlines(linesAddedToOrder);
+                            this.showErrorProductComboMessage();
+                            break;
+                        }
                     }
+                }
+            }
+
+            _mountComboProductOptions(productToAdd) {
+                return {
+                    price: productToAdd.price,
+                    quantity: productToAdd.quantity,
+                    merge: false,
+                };
+            }
+
+            _isDuplicateItemCategoryBehavior(categoryBehavior) {
+                return categoryBehavior === "duplicate_item";
+            }
+
+            _insertDuplicateOrderline(productToAdd, linesAddedToOrder) {
+                const productQuantity = productToAdd.quantity;
+                const lineNewPrice =
+                    this.selectedOrderline.get_unit_price() - productQuantity * 0.01;
+                this.selectedOrderline.set_quantity(1);
+                this.selectedOrderline.set_unit_price(lineNewPrice);
+                for (let i = 0; i < productQuantity; i++) {
+                    const clonedOrderline = this.selectedOrderline.clone();
+                    this.order.add_orderline(clonedOrderline);
+                    clonedOrderline.set_quantity(1);
+                    clonedOrderline.set_unit_price(0.01);
+                    linesAddedToOrder.push(clonedOrderline);
+                }
```

Comment:
Como os dados dos produtos devem ser iguais para todas as duplicatas, acho melhor adicionar só uma linha com a quantidade de duplicatas correspondente. Dessa forma tira a complexidade de mais um for no algorítimo.